### PR TITLE
STYLE: Further wrapping refactorings for ease of understanding

### PR DESCRIPTION
**Pull request by hjmjohnson**
Testing changes to wrapping interface.  This is a first pass at flattening the convoluted wrapping structure that heavily depends on global variables.

## PR Checklist
- [X] No  ** EXTERNAL ** [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Comment by hjmjohnson** [comment]
NOTE: I have also tested building the AnisotropicDiffusionLBR as a stand-alone application, and it seems to build the wrapping directories.

File: Wrapping/Generators/SwigInterface/CMakeLists.txt
Comment (hjmjohnson) [inline Wrapping/Generators/SwigInterface/CMakeLists.txt:396]: NOTE: No 'S; at the end of ITK_STUB_PYI_FILE.  The variable name should have been ITK_STUB_PYI_FILES.

File: Wrapping/Generators/CMakeLists.txt
Comment (dzenanz) [inline Wrapping/Generators/CMakeLists.txt]: Commit message does not match this addition.

File: Wrapping/Generators/Python/CMakeLists.txt
Comment (dzenanz) [inline Wrapping/Generators/Python/CMakeLists.txt:296]: This commit seems to move code from one place to another, but commit message implied only removal.

File: Wrapping/TypedefMacros.cmake
Comment (dzenanz) [inline Wrapping/TypedefMacros.cmake:332]: A bunch of these small commits should be squashed together.

File: Wrapping/Generators/CMakeLists.txt
Comment (hjmjohnson) [reply Wrapping/Generators/CMakeLists.txt]: Fixed and squashed several commit together.

File: Wrapping/TypedefMacros.cmake
Comment (hjmjohnson) [reply Wrapping/TypedefMacros.cmake:329]: Done.

File: Wrapping/Generators/Python/CMakeLists.txt
Comment (hjmjohnson) [reply Wrapping/Generators/Python/CMakeLists.txt:296]: Fixed by squashing and changing several commit messages.

**Comment by hjmjohnson** [comment]
@dzenanz Can you please help me identify why this file is not visible?

```txt
CMake Error at /usr/local/Cellar/cmake/3.22.2/share/cmake/Modules/ExternalData.cmake:642 (message):
  Data file referenced by argument

    DATA{/test/Input/BrainProtonDensitySlice.png}

  does not lie under the top-level source directory

    /Users/runner/work/1/s

Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.22.2/share/cmake/Modules/ExternalData.cmake:542 (_ExternalData_arg)
  /usr/local/Cellar/cmake/3.22.2/share/cmake/Modules/ExternalData.cmake:355 (ExternalData_expand_arguments)
  CMake/ITKModuleTest.cmake:73 (ExternalData_add_test)
  CMake/ITKModuleTest.cmake:146 (itk_add_test)
  Modules/Filtering/ImageGrid/wrapping/test/CMakeLists.txt:6 (itk_python_add_test)


-- Configuring incomplete, errors occurred!
See also "/Users/runner/work/1/s-build/CMakeFiles/CMakeOutput.log".
See also "/Users/runner/work/1/s-build/CMakeFiles/CMakeError.log".

```

**Comment by dzenanz** [comment]
The problem is probably the leading `/`. Try `DATA{test/Input/BrainProtonDensitySlice.png}`.

**Comment by hjmjohnson** [comment]
@dzenanz Thanks!  I was looking in the wrong place. Your help focused my attention on the misspelled variable name that needed fixing.  Updates pushed.  Fingers crossed that this builds completely.

**Comment by hjmjohnson** [comment]
@dzenanz @thewtex  It would be really nice to have this PR merged before the weekend so that I can focus on documenting and fixing the typehints code.

Tracing the generation of files is much easier after these changes.

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
Matt should take a look too.

**Comment by hjmjohnson** [comment]
> 

Ok.  @thewtex   If you approve, will you please press the "Rebase and Merge" button after reviewing?

File: CMake/ITKModuleExternal.cmake
Comment (thewtex) [inline CMake/ITKModuleExternal.cmake:68]: This should still be `ITK_DIR` for wrapping external modules. `ITK_BINARY_DIR` is only set when  building `ITK`.

**Review by thewtex — CHANGES_REQUESTED** [review]
As noted inline -- we need to continue to use `ITK_DIR` instead of `ITK_BINARY_DIR`.

A few intentions to keep in mind for the future, in spite of removing the other SWIG-based language generators:

1. We would like to replace SWIG with pybind11 at some point.
2. We would like to add WASM-based bindings.

These could be considered new "generators" or some similar compartmentalization architecture that will help our ability to reason about, scale, and maintain the codebase.

File: CMake/ITKModuleExternal.cmake
Comment (hjmjohnson) [reply CMake/ITKModuleExternal.cmake]: @thewtex  Thanks for catching this.  I did a test with an external build of AnisotropicDiffusionLBR with wrapping turned on.  It seemed to *not fail catastrophically* so I assumed the refactoring was OK.


I have removed these changes.

**Comment by hjmjohnson** [comment]
> A few intentions to keep in mind for the future, in spite of removing the other SWIG-based language generators:
> 
> 1. We would like to replace SWIG with pybind11 at some point. A transition period where both are alternatively possible in the code base will help.
> 2. We would like to add WASM-based bindings.
> 
> These could be considered new "generators" or some similar compartmentalization architecture that will help our ability to reason about, scale, and maintain the codebase.

I am very excited to consider pybind11.  SWIG does not seem to be long-term scalable solution for close python integration.

**Review by thewtex — APPROVED** [review]
@hjmjohnson please make sure to test building a remote module externally before merging.

**Comment by hjmjohnson** [comment]
@thewtex   Thanks!  I did test as a remote module:

```
git clone https://github.com/InsightSoftwareConsortium/ITKAnisotropicDiffusionLBR.git
cmake -G Ninja -DITK_DIR=/Users/johnsonhj/Dashboard/src/ITK/cmake-debug-PythonDebugWrapping-build -S ITKAnisotropicDiffusionLBR -B ADLBR-bld
ninja -C ADLBR-bld

....

Built without failure. 

```

**Pull request merged**
